mbed_lstools/linux.py

```diff
@@ -104,3 +110,52 @@ def _hex_ids(self, dev_list):
             match = self.nlp.search(dl)
             if match:
                 yield match.group("usbid"), _readlink(dl)
+
+    def _sysfs_block_devices(self, block_devices):
+        device_names = { os.path.basename(d): d  for d in block_devices }
+        sysfs_block_devices = set(os.listdir(SYSFS_BLOCK_DEVICE_PATH))
+        common_device_names = sysfs_block_devices.intersection(set(device_names.keys()))
+        result = {}
+
+        for common_device_name in common_device_names:
+            sysfs_path = os.path.join(SYSFS_BLOCK_DEVICE_PATH, common_device_name)
+            full_sysfs_path = os.readlink(sysfs_path)
+            path_parts = full_sysfs_path.split('/')
+
+            end_index = -1
+            for index, part in enumerate(path_parts):
+                if self.udp.search(part):
+                    end_index = index
+                    break
+
+            if end_index == -1:
+                logger.debug('Did not find suitable usb folder for usb info: %s', full_sysfs_path)
+                continue
+
+            usb_info_rel_path = path_parts[:end_index + 1]
+            usb_info_path = os.path.join(SYSFS_BLOCK_DEVICE_PATH, os.sep.join(usb_info_rel_path))
+
+            vendor_id = 'unknown'
+            product_id = 'unknown'
```

Comment:
I would like to use `None` here as well. I don't like string matching on "unknown" as an API.

Reply:
Agreed, how about I move the `None` - > `"unknown"` translation to the CLI layer?